rts/lra/lra-service-base/src/main/java/io/narayana/lra/LRAUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package io.narayana.lra;
+
+import io.narayana.lra.logging.LRALogger;
+import org.eclipse.microprofile.lra.client.GenericLRAException;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+
+public abstract class LRAUtils {
```

Comment:
I'm not keen on Utils type classes, they become a dumping ground for miscellaneous bits of code people think may or may not be generally useful.

Reply:
@mmusgrov generally true. But these methods are needed both in coordinator and in the client. They are all stateless static methods. All they do is transformations of the String and URL variants of lraId. Maybe we can change name to something different? Maybe `LraIdTransformator`?

Reply:
@xstefank I see that you force pushed a change that removed Utils making me think that you had accepted my proposal. But now I see that all you did was move the code into a different class (LRAIdConverter.java) and hide the change amongst a load of other changes. That was very sneaky, especially as I had already studied this large PR so assumed that since your force push removed Utils that the new commit was in response to this comment so I did not re-review the changes. Very poor form.

Reply:
@mmusgrov that was my mistake. I forgot to include that as part of the force push and I didn't expect this PR to be merged so soon prioritizing other work. Sorry for confusion, I will make another PR tomorrow amending these changes